news/4645-use-conda-index

```diff
@@ -0,0 +1,20 @@
+### Enhancements
+
+* Depend on standalone conda-index instead of bundled indexing code.
+
+### Bug fixes
+
+* <news item>
+
+### Deprecations
+
+* `conda-build index --file` is no longer supported.
+* `conda-build index --hotfix-source-repo` is removed.
```

Comment:
```suggestion
* `conda-build index --file` is no longer supported and is pending deprecation. (#4690)
* `conda-build index --hotfix-source-repo` is no longer supported and is pending deprecation. (#4690)
```

Reply:
Trying to communicate that `--file` no longer does anything, and `--hotfix-source-repo` produces an "unknown command line argument" error.

Reply:
Per CEP9 we shouldn't be removing things outright. We should update the code to allow these arguments but raise deprecation warnings with instructions on what to do instead.

Reply:
Doesn't CEP9 only apply to conda?

Reply:
This script is probably run in a non-interactive cron job. We could print a warning and fail to do what the user expects, since they are not actually there to observe the deprecation warning. Instead the current code exits.